src/motd.rs

```diff
@@ -25,23 +27,43 @@ impl RegexMatches {
     }
 }
 
+pub struct Counters {
+    tcp: Stats,
+    udp: Stats,
+}
+
+impl Counters {
+    pub fn new(tcp: Stats, udp: Stats) -> Self {
+        Counters {
+            tcp,
+            udp,
+        }
+    }
+}
+
 pub struct Motd {
     start_date_regex: RegexMatches,
     uptime_regex: RegexMatches,
+    counters_regex: RegexMatches,
     start_date: DateTime<Local>,
+    counters: Counters,
     template: String,
 }
 
 impl Motd {
-    pub fn new(template: String) -> Motd {
+    pub fn new(template: String, counters: Counters) -> Motd {
         let start_date_regex = Regex::new(r"(?i)\{\{\s*start_date\s*\}\}")
             .expect("Failed to compile start_date regex");
         let uptime_regex = Regex::new(r"(?i)\{\{\s*uptime\s*\}\}")
             .expect("Failed to compile uptime regex");
+        let counters_regex = Regex::new(r"(?i)\{\{\s*counters\s*\}\}")
```

Comment:
I like how you implemented counters. There is 1 question: should we add a variable per each counter or leave it as "counters" for all counters?

Reply:
Well, I can't decide on it. Who want to know only udp incoming or tcp outgoing.

Reply:
Separate counters give more freedom. It will be possible to specify 'Tcp: incoming {...}, outgoing {...}, Udp: incoming {...}, outgoing {...}' as a pattern via CLI, which seems is better than just a hardcoded string.